src/main/java/com/nextcloud/android/sso/helper/Retrofit2Helper.java

```diff
@@ -99,6 +101,19 @@ public boolean isCanceled() {
             public Request request() {
                 throw new UnsupportedOperationException("Not implemented");
             }
+
+            private Response<T> convertExceptionToResponse(int statusCode, String errorMessage) {
+                ResponseBody body = ResponseBody.create(null, errorMessage);
+                return Response.error(
+                        body,
+                        new okhttp3.Response.Builder()
+                                .body(body)
+                                .code(statusCode)
+                                .message(errorMessage)
+                                .protocol(Protocol.HTTP_1_1)
+                                .request(new Request.Builder().url("http://localhost/" + nextcloudRequest.getUrl()).build())
```

Comment:
Is this hardcoded localhost correct?

Reply:
It's never used anywhere (it's just needed for the error message). I copied this from the reotrofit lib and they are using localhost as well (hardcoded).